subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r81/CompositeBuildCrossVersionSpec.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.tooling.r81
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.tooling.model.gradle.GradleBuild
+
+@ToolingApiVersion(">=8.2")
+@TargetGradleVersion('>=8.2')
+class CompositeBuildCrossVersionSpec extends ToolingApiSpecification {
```

Comment:
❌  missing test: what happens when you call `getBuildTreePath()` on older Gradle versions? 

Right now, I assume it's an UnsupportedOperationException. It should be captured here.

I'm pretty sure we can mixin default values for older versions (`getPath()` or similar). Although I'm not sure, that would help anyone. Have a look at this method: https://github.com/gradle/gradle/blob/d161c6fb52e7f9662376b7b8664c6075694449c4/subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/connection/HasCompatibilityMapping.java#L45

Reply:
I'd got with UnsupportedMethodException (as is now).
I don't think lying or providing info that we don't have is useful.